### PR TITLE
Add git-retry-install.sh script

### PR DESCRIPTION
**Pull request by e-kov**
Now it can be used in the Dockerfiles like this

```
RUN curl https://raw.githubusercontent.com/elastio/devx/master/scripts/git-retry-install.sh | bash
ENV PATH=/usr/local/bin:$PATH
```

### CODE REVIEW

**Review by valdavasquez — APPROVED** [review]

**Review by Veetaha — APPROVED** [review]
I thought we would match the error message to define if the error is retriable or not as in [here](https://chromium.googlesource.com/infra/infra/+/b9f6e35d2aa1ce9fcb385e414866e0b1635a6c77/go/src/infra/tools/git/retry_regexp.go#17)

though it works either way

File: scripts/git-retry-install.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (e-kov) [reply scripts/git-retry-install.sh]: Moved all texts from the lines below to the lines above

File: scripts/git-retry-install.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# The real git is always in the '/usr/bin/'. Our retry wrapper will be in the '/usr/local/bin/'.
```
[the remark this answers is not in the document]
Comment (e-kov) [reply scripts/git-retry-install.sh:2]: > I suggest clarifying why we make such an assumption

Checked manually on all supported distros

File: scripts/git-retry-install.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    ret=$?
+    # don't retry if git command is not about network, like 'git log' or 'git diff' or
+    # getting branch name like in my .bashrc 'rev-parse --abbrev-ref HEAD'
+    cmd=$(echo "$@" | awk '{ print $1 }')
+    case $cmd in
+	    clone | commit | merge | rebase | tag | fetch | pull | push) ;;
+	    *) break ;;
+    esac
+    # git returns 1 in case if command is incorrect, for other cases
+    # it retrns 128 or 130. So, we should not retry on 1.
```
[the remark this answers is not in the document]
Comment (e-kov) [reply scripts/git-retry-install.sh]: thnks. fixed

File: scripts/git-retry-install.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+RETRIES=15
+DELAY=3
+for (( i=0; i<=$RETRIES; i++ )) ; do
+    $GIT "$@"
+    ret=$?
+    # don't retry if git command is not about network, like 'git log' or 'git diff' or
+    # getting branch name like in my .bashrc 'rev-parse --abbrev-ref HEAD'
+    cmd=$(echo "$@" | awk '{ print $1 }')
+    case $cmd in
+	    clone | commit | merge | rebase | tag | fetch | pull | push) ;;
```
[the remark this answers is not in the document]
Comment (e-kov) [reply scripts/git-retry-install.sh]: fixed.

**Pull request merged**